pytarallo/Item.py

```diff
@@ -1,60 +1,62 @@
 from typing import Optional, Dict, Any, List
+from pytarallo.Product import Product
 
 
 class Item:
     """This class implements a pseudo-O(R)M"""
     code: Optional[str]
     features: Dict[str, Any]
+    product: Product
     contents: List[Any]  # Other Item objects, actually
-    location: Optional[str]
-    path: List[str]
-    parent: Optional[str]
+    location: List[Any]
 
     def __init__(self, data: dict = None):
         """
-        Items are generally created by the get_item method
-        But could be created somewhere else and then added to the
-        database using the add_item method
+        Items are created by the get_item method
         params: data: a dict() containing the item's data.
         """
         self.code = None
         self.features = {}
+        self.product = None
+        # http://localhost:8080/v2/items/R777?separate
         self.contents = list()
-        self.location = None
-        self.path = []
-        self.parent = None
+        self.location = list()
 
         if data is not None:
-            self.code = data['code']
-
-            for k, v in data['features'].items():
-                # setattr(self, k, v)
-                self.features[k] = v
-
             # setup path and location
-            # location: the most specific postion where the item is located, e.g. "Table"
+            # location: the most specific position where the item is located, e.g. "Table"
             # path: a list representing the hierarchy of locations the item, e.g. ["Polito", "Chernobyl", "Table"]
             # the tarallo server returns an JSON object containing only the "location" attribute, which corresponds
             # to the path as defined above
+            if data.get('code') is not None:
```

Comment:
Smart use of `get` to not  write he more verbose `'code' in data and data['code']`.
In Python, `None` is falsy, so we can omit the check, especially if code is a string (don't do this if code can be an empty list or other falsy variables):
```suggestion
            if data.get('code'):
```
You can apply this suggestion to other similar lines as well.